sopel/config/core_section.py

```diff
@@ -1213,6 +1213,43 @@ def homedir(self):
 
     """
 
+    tls_ciphers = ValidatedAttribute(
+        'tls_ciphers',
+        default=":".join(
+            [
+                "ECDHE-ECDSA-AES128-GCM-SHA256",
+                "ECDHE-RSA-AES128-GCM-SHA256",
+                "ECDHE-ECDSA-AES256-GCM-SHA384",
+                "ECDHE-RSA-AES256-GCM-SHA384",
+                "ECDHE-ECDSA-CHACHA20-POLY1305",
+                "ECDHE-RSA-CHACHA20-POLY1305",
+                "DHE-RSA-AES128-GCM-SHA256",
+                "DHE-RSA-AES256-GCM-SHA384",
+            ]
+        )
+    )
+    """The cipher suites enabled for SSL/TLS connections.
+
+    :default: ``ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384``
+
+    The default is the `Mozilla Intermediate configuration`_ as of 2022-02.
+    This parameter is in OpenSSL format, see the `ciphers manual page`_ for
+    your version. This setting cannot be used to configure TLS 1.3.
+
+    .. _`Mozilla Intermediate configuration`: https://wiki.mozilla.org/Security/Server_Side_TLS
+    .. _`ciphers manual page`: https://www.openssl.org/docs/manmaster/man1/ciphers.html
```

Comment:
[Anonymous hyperlinks](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#anonymous-hyperlinks) are fine, even preferred (since it eliminates the chance of creating reference-name-collision errors later).

Also note preference for a prose-form month/year, & fixing of a comma splice.

```suggestion
    The default is the `Mozilla Intermediate configuration`__ as of February 2022.
    This parameter is in OpenSSL format; see the `ciphers manual page`__ for
    your version. This setting cannot be used to configure TLS 1.3.

    .. __: https://wiki.mozilla.org/Security/Server_Side_TLS
    .. __: https://www.openssl.org/docs/manmaster/man1/ciphers.html
```

Finally: What does it mean that "This setting cannot be used to configure TLS 1.3"?

Reply:
TIL that exists, thanks. I wish it had [1] style footnotes like markdown though, I could see this getting annoying when there's more than a couple links or if any are duplicated.
The TLS 1.3 comment is from the python `ssl` docs: https://docs.python.org/3/library/ssl.html#ssl.SSLContext.set_ciphers and https://docs.python.org/3/library/ssl.html#tls-1-3
This is because TLS 1.3 replaces specific cipher suites where you needed to list every permutation (e.g. "ECDHE-ECDSA-AES128-GCM-SHA256") with a caps-like system ("I support enc=AES128/CHACHA20, mode/mac=GCM/POLY1305, hash=SHA256").

Reply:
There are _many_ different ways to make links in rST, including a footnote-y style where you give <tt>\`link text <_target-name>\`_</tt> and then define the link target named `_target-name` later. The anonymous style is just much simpler to write and it's more than adequate for our purposes where the intended use is as rendered HTML.

All of that said… I honestly prefer Markdown's link syntaxes. But Sphinx is built for rST, so we're kind of stuck with that in docstrings for now.

Would it be overly complicated to expose a separate option for TLSv1.3 cipher suites, then? I don't deny that Python's defaults are probably better than whatever I'd come up with independently, but I can imagine there could be cases where the defaults Just Don't Work™ for certain servers.

Reply:
Those settings aren't exposed in Python, so rather inexpedient to implement :smile_cat:
If you needed tight control over the ciphers used, you could set the *max* tls version and tweak 1.2, but barring some "the server selects chacha20 and has a broken implementation" nonsense there's no reason to muck with it anyway.

Reply:
Presuming that if Python does expose them in the future, we'll still need to differentiate between 1.3 and 1.2 ciphers because of the different formats… want to rename the option so it's clear that it's only for 1.2?

Reply:
I figure there's a substantial chance openssl or python ends up shoving 1.3 config into the same cipher spec string, and I'd rather carry `tls_ciphers`+`tls_caps` cruft than `tls12_and_older_ciphers`+"wait no actually it's all of them" cruft